### PR TITLE
Implement crud Operations

### PR DESCRIPTION
**Pull request by rosemutai**
- Implemented a function for adding a new task to the list of existing tasks.
- Implemented a function for deleting a task.
- Implemented a function for editing task descriptions.
- Store tasks in local storage

### CODE REVIEW

File: src/modules/Todo.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const descriptionInput = document.getElementById('description-input');
+let allToDos = [];
+const todosList = document.getElementById('todos');
+
+const deleteToDoItem = (index) => {
+  allToDos.splice(index, 1);
+  todosList.innerHTML = '';
+  localStorage.setItem('alltasks', JSON.stringify(allToDos));
+};
```
Comment (ghost) [inline src/modules/Todo.js]: #### Good work displaying the tasks but there are some aspects you can improve
- [x] As stated in the project requirement, Kindly ensure that - By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).

image

- I tried clicking on the `three doted icon` on your design and I noted the tasks were being deleted but that could only be displayed after I refresh the page. Kindly try following [this](https://www.youtube.com/watch?v=AcUd-_Yjjqg) video to guide you on how to implement the delete functionality. Also note that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).
- I could not edit your tasks. Kindly ensure that all your tasks can be edited and change saved in the local storage as stated in the project requirement

image

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @rosemutai , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Github workflow
✅ No linter error


## Required Changes ♻️
- [x] If you have not deployed the project yet, I think it would be better to indicate that the `live demo` link is not available. Having broken links on your readme file hinders user experience: If users encounter broken links in the README file, they may become frustrated and give up on using the software. This can harm the adoption of the project and lead to a decrease in its popularity.

_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/Todo.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+const editToDoList = (index, newValue) => {
+  allToDos[index].description = newValue;
+};
+
+const toggleCompletedTask = (index) => {
+  allToDos[index].completed = !allToDos[index].completed;
+};
```
Comment (ishimwezachee) [inline src/modules/Todo.js]: - [ ]  As previous reviewer suggested, you still have some tasks that are true by default. while they should be true only if the user check the tasks as done. 
image

File: src/modules/Todo.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+let allToDos = [];
+const todosList = document.getElementById('todos');
+
+const deleteToDoItem = (index) => {
+  allToDos.splice(index, 1);
+  todosList.innerHTML = '';
+  localStorage.setItem('alltasks', JSON.stringify(allToDos));
+};
+
+const editToDoList = (index, newValue) => {
```
Comment (ishimwezachee) [inline src/modules/Todo.js]: - [ ]  As previous reviewer suggested the functions for `editing `and `deleting `the task should work as expected. at this moment a user can not edit or delete a created task. pleasure check the requirements those functions as suggested as well.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @rosemutai , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
image


 ## Highlights
✅  Project is running successfully 
✅ No linter error

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [OPTIONAL ] I recommend you to use the correct gitflow, for this  project is individual project, the base branch should  be `dev `branch not `main `branch 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @rosemutai , 

## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Linters are passing ✔️ 
- README is professional 👍 
- Great UI 🥇 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**